### PR TITLE
Support target session attrs

### PR DESCRIPTION
**Pull request by bpicolo**
This is a remake of #714 that pulls in much related code, but also has a couple bugfixes for SSL tests, and updates some related documentation (for how to run multi-pg tests locally).

### CODE REVIEW

**Comment by bpicolo** [comment]
Closing: this was intended for my fork

File: conn.go [hunk 1]
```diff
@@ -325,43 +346,61 @@ func DialOpen(d Dialer, name string) (_ driver.Conn, err error) {
...
+		if timeout, ok := o["connect_timeout"]; ok && timeout != "0" {
+			err = cn.c.SetDeadline(time.Time{})
+		}
+		panicking = false
+		return cn, err
 	}
-	panicking = false
-	return cn, err
+
+	// In most cases
```
Comment (bpicolo) [inline conn.go]: Forgot to finish this sentence.

There's some question about how we would want to return errors/ connections here. Currently this takes the path of least resistance and uses the latest connection / err

**Comment by costela** [comment]
Maybe I'm missing something, but why couldn't this have been done as an improvement to #714? Seems a bit impolite to take my initial code and just squash it under your name.

**Comment by bpicolo** [comment]
@costela probably could be! Happy to close this. I needed it myself and am not used to PRing other people’s forked PRs and GitHub doesn’t provide the best social interactivity there =]. Happy to figure out how to integrate / close this with you. Definitely appreciate the work you put in

**Comment by costela** [comment]
Sure, totally understandable. Thanks for the clarification!
I'd suggest the following: open PRs for your improvements against my fork and I'll rebase the original PR with your additional commits. This way we can work together on keeping this fork relatively up-to-date against the original repo.

And if it turns out we have fundamentally different ideas about how this feature should work, you can always open a separate PR in the future again.

**Pull request closed without merging**